lib/rubocop/cop/layout/end_alignment.rb

```diff
@@ -72,6 +72,15 @@ class EndAlignment < Cop
         include EndKeywordAlignment
         include RangeHelp
 
+        def on_kwbegin(node)
+          align_with = {
+            keyword: node.loc.begin,
```

Comment:
Is this right? The other callbacks, like `on_class` use `node.loc.keyword` but `Parser::Source::Map::Collection` doesn't respond to `keyword`. I'd really appreciate feedback here, I'm not at all confident this is correct. Thanks.

Reply:
@jaredbeck I've been [asking](https://github.com/whitequark/parser/issues/357) the maintainers of `parser` if the source map can be changed, but they are not interested. They suggested we patch it in RuboCop instead.

I think patching the source map is probably better, since there are other cops that are affected by this as well.